gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/internal/mfa/MFAEnrollStep.java

```diff
@@ -54,6 +53,7 @@ public void execute(RoutingContext routingContext, AuthenticationFlowChain flow)
         final Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
         final MfaFilterContext context = new MfaFilterContext(routingContext, client, factorManager);
         if (hasFactors(client, factorManager)) {
+            context.setDefaultFactorWhenApplied(ruleEngine);
```

Comment:
**suggestion**: I do prefer to pass the engine in the context constructor instead of as param of this method. 
**quetion**: Why do we have to set the default factor id here ?

Reply:
The diagram flow: if a user wants to enroll a factor that does not match the rule it falls back to the default factor.